### PR TITLE
Added an additional mode to 'metarule2_explosions' function.

### PR DESCRIPTION
**Pull request by FakelsHub**
Sets a permanent (reset to default after loading the game) radius of the explosion for rockets or grenades.

`LoadGameHook::OnGameReset() += ResetExplosionRadius;`
It is necessary to check the reset values, I'm not sure that it is code working)

### CODE REVIEW

**Comment by NovaRain** [comment]
>For what reason did you send a review request for phobos? does something bother you in the code?

Because phobos has much more experience on explosion related scripting than me, I think he may have better opinion on whether this additional mode is necessary, or the current set_attack_explosion_radius is enough.

**Comment by FakelsHub** [comment]
The additional mode sets a separate radius for grenades and rocket/dynamite.
I think this mode will be more useful than constant monitoring of explosions and the used of the temp function set_attack_explosion_radius.

ps: do you have any connection with Phobos? he is no longer in touch for two months.

File: sfall/Modules/Explosions.cpp [hunk 1]
```diff
@@ -30,6 +31,7 @@ namespace sfall
 {
 
 static bool lightingEnabled = false;
+static bool explosionsMetarule = false;
```
Comment (phobos2077) [inline sfall/Modules/Explosions.cpp]: @FakelsHub , can you give a better name for this flag, like `needResetExplosionsAfterAttack`?

File: sfall/Modules/Explosions.cpp [hunk 2]
```diff
@@ -180,27 +191,33 @@ int _stdcall ExplosionsMetaruleFunc(int mode, int arg1, int arg2) {
...
 			}
 			break;
-		}
+		case EXPL_STATIC_EXPLOSION_RADIUS:
+			if (arg1 > 0) set_expl_radius_grenade = arg1;
+			if (arg2 > 0) set_expl_radius_rocket = arg2;
+			SetExplosionRadius(set_expl_radius_grenade, set_expl_radius_rocket);
+			break;
+		default:
+			return -1;
```
Comment (phobos2077) [inline sfall/Modules/Explosions.cpp]: Consider printing error to the debug log while returning -1. I think some of the other functions had those.

**Comment by phobos2077** [comment]
Overall the code looks fine, I hope you will test it though. Good thing this is reset in the `GameReset` event :+1:

**Comment by FakelsHub** [comment]
Updated.

**Comment by FakelsHub** [comment]
what with this PR, why not accept?

**Comment by NovaRain** [comment]
I tested the new mode a bit, seems working fine. I use HOOK_COMBATDAMAGE to call ```set_explosion_radius(5, 6);``` before every attack, so all my grenades and rockets have super large radius.

**Comment by phobos2077** [comment]
@FakelsHub  are you deleting the old commits and then repeat the changes?

**Comment by FakelsHub** [comment]
Ничего не удаляю, делаю push с опцией -force вроде так, в общем у меня git tortorise ставлю галочку и пушу, а что там происходит в системе git мне уже не интересно. :-)

**Comment by phobos2077** [comment]
@FakelsHub какие соображения, кури дизассемблированный код) По мне так фича не нужна. Не вижу ничего плохого в пересборке скрипта двери..

Если лень писать комментарий, хотябы что-то вроде "Code improvement" будет лучше чем удалять комиты.

**Pull request merged**